lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/TransactionSpec.hs

```diff
@@ -478,6 +506,27 @@ goldenTestStdTx tl keystore inps outs bytes' = it title $ do
   where
     title = "golden test mkStdTx: " <> show inps <> show outs
 
+goldenTestDelegationCertTx
+    :: forall t k. (t ~ Jormungandr, HasCallStack)
+    => TransactionLayer t k
+    -> (Address -> Maybe (k 'AddressK XPrv, Passphrase "encryption"))
+    -> PoolId
+    -> (k 'AddressK XPrv, Passphrase "encryption")
+    -> [(TxIn, TxOut)]
+    -> [TxOut]
+    -> ByteString
+    -> SpecWith ()
+goldenTestDelegationCertTx tl keystore pool (accountXPrv, pass) inps outs bytes' = it title $ do
+    let res = mkDelegationCertTx tl pool (accountXPrv, pass) keystore inps outs
+    let sealed = getSealedTx . snd <$> res
+    sealed `shouldBe` (Right $ unsafeFromHex bytes')
+    & counterexample ("poolId = " <> showHex (getPoolId pool))
```

Comment:
**[question]** 
Does this `counterexample` actually works? We aren't even inside a property here, though `shouldBe` might do the trick here. How does a failure look like here?

Reply:
Ehrm… Yes! c.f. https://github.com/input-output-hk/cardano-wallet/pull/1059#discussion_r350683421